src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/StructBuilder.java

```diff
@@ -136,6 +136,14 @@ private String findAnonymousStructName(GroupLayout parentLayout, GroupLayout lay
 
     @Override
     public void addVar(String javaName, String nativeName, MemoryLayout layout, Class<?> type) {
+        try {
+            structLayout.byteOffset(elementPaths(nativeName));
```

Comment:
Now that I think about it - calling byteSize on the structLayout should be enough. I also think that, if we do this, we can remove the logic in OutputFactory which checks whether size is available.

Reply:
hmm.. calling byteSize on structLayout elsewhere? i.e., before addVar being called?

Reply:
hmm.. byteSize() does not result in exception being thrown (the previous struct field has undermined size because of unsized array at the end. but current offending field has proper size). Am I missing something?

Reply:
I mean byteSize on `structLayout`. If there is any variable-length array, structLayout (which is supposedly storing the _enclosing_ layout of all nested anon structs inside it as well) should have its size undefined too.